interface/main/holidays/import_holidays.php

```diff
@@ -0,0 +1,177 @@
+
+        <p class='text'>
+        <table border='1' cellpadding='4'>
+            <tr bgcolor='#dddddd' class='dehead'>
+                <td align='center' colspan='2'>
+                    <?php echo xlt('CSV'); ?>
+                </td>
+            </tr>
+            <tr>
+                <td class='detail' nowrap>
+                    <?php echo htmlspecialchars(xl('CSV File')); ?>
+                    <input type="hidden" name="MAX_FILE_SIZE" value="350000000" />
+                </td>
+                <td class='detail' nowrap>
+                    <input type="file" name="form_file" size="40" />
+                </td>
+            </tr>
+            <tr>
+                <td class='detail' nowrap>
+                    <?php echo xlt('File on server (modification date)'); ?>
+                </td>
+                <td class='detail' nowrap>
+                    <?php
+                    if(!empty($csv_file_data)){?>
+
+                        <?php $path=explode("/",$holidays_controller->get_target_file());?>
+                        <?php $filename=$path[count($path)-1];?>
+                        <?php unset($path[count($path)-1]);?>
+
+                        <?php
+                        // build the path to the for the download request
+                        $phppagepath = "http://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]?filename=";?>
+                        <a href="#" onclick='window.open("<?php echo $phppagepath; echo $holidays_controller->get_target_file();?>&download_file=1")'><?php echo $csv_file_data['date'];?></a>
+                    <?php }else{
+                        echo htmlspecialchars(xl('File not found'));
+                    }?>
+                </td>
+            </tr>
+
+        <tr bgcolor='#dddddd'>
+                <td align='center' class='detail' colspan='2'>
+                    <input type='submit' name='bn_upload' value='<?php echo xlt('Upload / Save ') ?>' />
+                </td>
+            </tr>
+        </table>
+        </p>
+</form>
+        <table border='1'>
+
+        <tr >
+            <td >
+                <form method='post' action='import_holidays.php' onsubmit='return top.restoreSession()'>
+                    <input type='submit' name='import_holidays' value='<?php echo xlt('Import holiday events ') ?>'></br>
+
+                </form>
+            </td>
+
+            <td>
+                    <?php echo xlt('CSV to calendar_external table'); ?></br>
+                    If  the csv has been uploaded please click on "parse and insert" button</br>
+                    **NOTE: clicking on the button will remove all the existing rows in the calendar external table
+                </td>
+        </tr>
+            <tr >
+                <td >
+                    <form method='post' action='import_holidays.php' onsubmit='return top.restoreSession()'>
+                        <input type='submit' name='sync' value='<?php echo xlt('Syncronize ') ?>' /></br>
```

Comment:
This is within an html attribute, so should use xla() and not xlt():
http://www.open-emr.org/wiki/index.php/Codebase_Security#SQL-Injection_and_Cross-Scripting_Prevention
(see step 4 example, and always feel free to ask if need more detailed explanation on differences between xla/xlt and attr/text functions)